src/osa/paths.py

```diff
@@ -397,10 +399,100 @@ def create_longterm_symlink(cherenkov_job_id: str = None):
     else:
         log.warning(f"Job {cherenkov_job_id} (lstchain_cherenkov_transparency) did not finish successfully.")
 
+
 def dl1_datacheck_longterm_file_exits() -> bool:
     """Return true if the longterm DL1 datacheck file was already produced."""
     nightdir = utils.date_to_dir(options.date)
     longterm_dir = Path(cfg.get("LST1", "LONGTERM_DIR"))
     longterm_file = longterm_dir / options.prod_id / nightdir / f"DL1_datacheck_{nightdir}.h5"
     return longterm_file.exists()
 
+
+def convert_dec_string(dec_str: str) -> float:
+    """Return the declination angle in degrees corresponding to a 
+    given string of the form "dec_XXXX" or "dec_min_XXXX"."""
+    # Split the string into parts
+    parts = dec_str.split('_')
+
+    # Extract the sign, degrees, and minutes
+    sign = 1 if 'min' not in parts else -1
+    degrees = int(parts[-1])
+
+    # Calculate the numerical value
+    dec_value = sign * (degrees / 100)
+
+    return dec_value
+
+
+def get_corresponding_string(list1: list, list2: list) -> dict:
+    """Return a dictionary created from two given lists."""
+    corresponding_dict = {}
+    for index, element in enumerate(list2):
+        corresponding_dict[element] = list1[index]
+    return corresponding_dict
+
+
+def get_latest_RF_model_path(dec_str: str) -> Path:
+    """Get the path of the most recent version of RF models for a given declination,
+    excluding the ones produced for the source-dependent analysis."""
+    BASE_MODELS = Path("/fefs/aswg/data/models/AllSky")
+    # make sure the RF models correspond to the current version of lstchain
+    current_version = get_major_version(utils.get_lstchain_version())
+    list_nodes = sorted(BASE_MODELS.rglob(f"*{current_version}*/{dec_str}"), key=os.path.getmtime)
+    
+    log.debug(f"Found len(list_nodes) paths with {current_version} corresponding to {dec_str}:")
+    for path in list_nodes:
+        log.debug(path)
+    
+    # remove from the list the models produced for the source-dependent analysis        
+    for i in list_nodes:
+        if "srcdep" in str(i):
+            list_nodes.remove(i)
+
+    return list_nodes[-1]
+        
+
+def get_RF_model(run_str: str) -> Path:
+    """Get the path of the RF model to be used in the DL2 production for a given run."""
+    run_catalog_dir = Path("/fefs/aswg/data/real/monitoring/RunCatalog")
+    run_catalog_file = run_catalog_dir / f"RunCatalog_{options.date}.ecsv"
+    run_catalog = Table.read(run_catalog_file)
+    run = run_catalog[run_catalog["run_id"]==int(run_str)]
+    
+    try:
+        target_name = run["source_name"]
+        source_coordinates = SkyCoord.from_name(target_name)
+        source_dec = source_coordinates.dec.value
+        
+    except TypeError:
+        source_dec = run["source_dec"][0]
```

Comment:
In the way it is fetched right now from the TCU database, this is not exactly the declination of the source, but the declination at which the telescope is "pointing". It should be possible to look for the target coordinates in the TCU database though.